publication-commons/src/test/java/no/unit/nva/publication/service/impl/MessageServiceTest.java

```diff
@@ -117,10 +138,59 @@ void shouldThrowUnauthorizedWhenAttemptingToDeleteMessageUserDoesNotOwn()
+    @ParameterizedTest
+    @ValueSource(classes = {
+        DoiRequest.class,
+        GeneralSupportRequest.class,
+        PublishingRequestCase.class})
+    void shouldAllowCuratorToDeleteMessageWhenTicketHasCorrectType(Class<? extends TicketEntry> ticketType) throws ApiGatewayException {
+        var publication = TicketTestUtils.createPersistedPublicationWithOwner(
+            PublicationStatus.PUBLISHED, owner, resourceService);
+        var ticket = TicketTestUtils.createPersistedTicket(publication, ticketType, ticketService);
+        var persistedMessage = messageService.createMessage(ticket, owner, randomString());
+
+        var doiCurator = randomUserInstance(AccessRight.MANAGE_DOI, owner.getCustomerId());
+        var supportCurator = randomUserInstance(AccessRight.SUPPORT, owner.getCustomerId());
+        var publishingCurator = randomUserInstance(AccessRight.MANAGE_PUBLISHING_REQUESTS, owner.getCustomerId());
+
+        var doiCuratorFromAnotherInstitution = randomUserInstance(AccessRight.MANAGE_DOI);
+        var supportCuratorFromAnotherInstitution = randomUserInstance(AccessRight.SUPPORT);
+        var publishingCuratorFromAnotherInstitution = randomUserInstance(AccessRight.MANAGE_PUBLISHING_REQUESTS);
+
+        if (ticketType == DoiRequest.class) {
+            assertDoesNotThrow(() -> messageService.deleteMessage(doiCurator, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(doiCuratorFromAnotherInstitution, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(publishingCurator, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(supportCurator, persistedMessage));
+        }
+
+        if (ticketType == GeneralSupportRequest.class) {
+            assertDoesNotThrow(() -> messageService.deleteMessage(supportCurator, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(supportCuratorFromAnotherInstitution, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(doiCurator, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(publishingCurator, persistedMessage));
+        }
+
+        if (ticketType == PublishingRequestCase.class) {
+            assertDoesNotThrow(() -> messageService.deleteMessage(publishingCurator, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(publishingCuratorFromAnotherInstitution, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(doiCurator, persistedMessage));
+            assertThrows(UnauthorizedException.class, () -> messageService.deleteMessage(supportCurator, persistedMessage));
+        }
+    }
```

Comment:
I would refactor this test to two tests, the one which allows deleting, and another one which not.

You can use following method source for test which tests allowance:

```
    public static Stream<Arguments> ticketTypeAccessRightsProvider() {
        return Stream.of(Arguments.of(DoiRequest.class, MANAGE_DOI),
                         Arguments.of(PublishingRequestCase.class, MANAGE_PUBLISHING_REQUESTS),
                         Arguments.of(UnpublishRequest.class, MANAGE_PUBLISHING_REQUESTS),
                         Arguments.of( GeneralSupportRequest.class, SUPPORT));
    }
```
Your test will look like this:
```
  @ParameterizedTest
    @MethodSource("ticketTypeAccessRightsProvider")
    void shouldAllowCuratorToDeleteMessageWhenTicketHasCorrectType(Class<? extends TicketEntry> ticketType,
                                                                   AccessRight accessRight) throws ApiGatewayException {
        var publication = TicketTestUtils.createPersistedPublicationWithOwner(
            PublicationStatus.PUBLISHED, owner, resourceService);
        var ticket = TicketTestUtils.createPersistedTicket(publication, ticketType, ticketService);
        var persistedMessage = messageService.createMessage(ticket, owner, randomString());
        var curator = randomUserInstance(accessRight, owner.getCustomerId());

        assertDoesNotThrow(() -> messageService.deleteMessage(curator, persistedMessage));
    }
```

Reply:
You need rule for unpublishRequest to this method to make test pass:
`private boolean isCuratorOnTicket(Message message, UserInstance userInstance)` 

I do not know the requirements here:
`case UnpublishRequest unpublishRequest-> true;
`

Reply:
Just an example :)